CONTRIBUTING.md

```diff
@@ -11,6 +11,30 @@ sign a Contributor License Agreement (CLA). For an explanation of
 why we ask this as well as instructions for how to proceed, see the
 [Microsoft CLA](https://cla.opensource.microsoft.com/).
 
+### Devcontainer / Github Codespaces
+
+The easiest way to start contributing is via our devcontainer. This container works both locally in visual studio code with docker-desktop/docker-for-mac as well as [Github Codespaces](https://github.com/features/codespaces). To open the project in vscode you will need the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers). For codespaces you will need to [create a new codespace](https://codespace.new/citusdata/citus).
+
+With the extension installed you can run the following from the command pallet to get started
+
+```
+> Dev Containers: Clone Repository in Container Volume...
+```
+
+In the subsequent popup paste the url to the repo and hit enter.
+
+```
+https://github.com/citusdata/citus
+```
+
+This will create an isolated Workspace in vscode, complete with all tools required to build, test and run the Citus extension. We keep this container up to date with the supported postgres versions as well as the exact versions of tooling we use.
+
+To quickly start we suggest splitting your terminal once to have two shells. The left one in the `/workspaces/citus`, the second one changed to `/data`. The left terminal will be used to interact with the project, the right one with a testing cluster.
+
+To get citus installed from source we run `make install -sj8` in the first terminal, adjust `j8` to suite to the number of cores you have available. Once installed you can start a Citus cluster in the second terminal via `citus_dev make citus`. The cluster will run in the background, and can be interacted with via `citus_dev`. To get an overview of the available commands.
```

Comment:
Probably nicer to add `export MAKEFLAGS="-j $(nproc)"` to bashrc

Reply:
Or probably better, our make file adding this: https://stackoverflow.com/a/60413199/2570866

Reply:
I have added it to .bashrc for now, and also use the same ENV for building postgres, to allow for higher parallelism on bigger boxes.

We can later evaluate if adding it to our makefile makes sense, since that's a change outside of the devcontainer scope. I tried to keep this change as isolated from the rest of the codebase as possible to make it easy for rubber stamping.